### ISSUE TITLE
Video's Not Playing

### ISSUE THREAD

**Issue by DereckR** [Firebottle/Firebot#316]
#### Describe your issue or feature request (details are helpful!):

Video's were having issues with volume in last release so I updated to newest version. Now the video's are not playing sometimes will but now they won't at all. Logs don't show any errors just doesn't play audio or the videos?

-----------------------------------------------------------------------
#### If you're reporting an issue, *please* fill out & do the following:

- What version of Firebot are you running?

4.0.1

- What are the steps to reproduce the issue?
1) Press button to play video

Is there any other way to share the zip do not want to be sharing my sound board JSON as it has file paths names and other personal info in them.

```
NOTE: A lot of issues can be solved by,
- Updating to the latest version
- Trying the troubleshooting steps on the wiki
```

**Comment by Firebottle** [comment]
Local video or youtube video?

**Comment by DereckR** [comment]
Local Video

**Comment by Firebottle** [comment]
Spoke with someone else having the same issue. They were able to get images and other overlay elements showing, but video would not play. Attaching the video that the other person was testing with.
[Crunk Batman.zip](https://github.com/Firebottle/Firebot/files/1198182/Crunk.Batman.zip)

**Comment by Firebottle** [comment]
@DereckR Could you try removing the overlay.html file from your OBS/Xsplit and then re-adding that back in for me?

**Comment by ThePerry-2015** [comment]
So just a small input here.. Check the url/path for the firebot.html overlay. It should be something like:
`file:///C:/Users/{usernamehere}/AppData/Roaming/Firebot/firebot-data/overlay/firebot.html`

**Comment by DereckR** [comment]
Duh of course OBS was point old location

**Comment by DereckR** [comment]
@ThePerry-2015 Well I changed it and still no video plays or audio definitely wasn't the fix.

**Comment by Firebottle** [comment]
Did you try toggling the OBS compatibility in the Firebot settings?

**Comment by imjeeves** [comment]
I'm having a similar issue with playing videos.  Here's what I've done to troubleshoot:

- Created a short mp4 video (about 10 seconds) from downloading a youtube video, cutting it up in Adobe Premiere, and exporting it using standard settings.
- Set the button in Firebot to play the local video using pretty standard settings.
- I changed the overlay compatibility to "Other" and opened up the firebot.html file within a web browser and tested the video button - I saw the video play in the browser but not in OBS
- I changed the overlay compatibility to "OBS" and did the same test - the video did *not* play in the browser nor did it play in OBS
- I went in to OBS to the Firebot source and unchecked "local file" and set the URL to: file:///C:/Users/[username]/AppData/Roaming/Firebot/firebot-data/overlay/firebot.html then ran through the same tests
- Overlay Compatibility to "Other" with "URL" Browser source (rather than local): Video played in both browser *and* OBS
- Overlay Compatibility to "OBS" with "URL" Browser source: Video played ONLY in OBS (not in browser)

This somewhat fixed my issue, except the "video location" option doesn't seem to have any affect.  I set it to upper left, and the video plays in upper left.  I set it to middle and the video plays in upper left.  I set it to bottom right and video plays in upper left - it never moves.  Not sure if that's a related issue or if it's something different and I should open another ticket for that.

**Comment by ThePerry-2015** [comment]
Ok, so I just retested this to be sure it works on my end.

**Overlay compatibility: OBS**
*  OBS overlay as local file: Local video, Position: Bottom Right, Size: 400x225 - **Works OK**.
*  OBS overlay as local file: YouTube video, Position: Bottom Right, Size: 400x225 - **Works OK**.
*  OBS as file:/// URL: Local video, Position: Bottom Right, Size: 400x225 - **Does not work, which is correct because of file paths**.
*  OBS as file/// URL: YouTube video, Position: Bottom Right, Size: 400x225 - **Works OK, which is to expect since it's not a locally hosted video**.
*  Chrome: Local video, Position: Bottom Right, Size: 400x225 - **Does not work, which is right since Chrome don't support the file paths used in OBS compatibility mode**.
*  Chrome:: YouTube video, Position: Bottom Right, Size: 400x225 - **Works OK, which is to expect since it's not a locally hosted video**.

** Overlay Compatibility: Other**
*  OBS as file:/// URL: Local video, Position: Bottom Right, Size: 400x225 - **Works OK**.
*  OBS as file/// URL: YouTube video, Position: Bottom Right, Size: 400x225 - **Works OK**.
*  OBS overlay as local file: Local video, Position: Bottom Right, Size: 400x225 - **Does not work, which is correct because of file paths**.
*  OBS overlay as local file: YouTube video, Position: Bottom Right, Size: 400x225 - **Works OK, which is to expect since it's not a locally hosted video**.
*  Chrome: Local video, Position: Bottom Right, Size: 400x225 - **Works OK**.
*  Chrome:: YouTube video, Position: Bottom Right, Size: 400x225 - **Works OK**.

I've now tested this on two different computers, even installed OBS and Firebot on my daughters brand new Acer Aspire V17 Nitro just check, and it worked just the same.
So I'm really not sure why this doesn't work on your end.

**Comment by DereckR** [comment]
I deleted them all and re added and they are working now.

**Comment by Firebottle** [comment]
@imjeeves see above. Let us know if the same thing fixes your issue also.

**Comment by imjeeves** [comment]
@Firebottle mine is a fresh install of Firebot - I've never used it until now.  So all of the buttons are brand new.

I was just watching another streamer friend of mine play with his firebot as well.  He was having the same issues that I was, and the same "fix" worked (changing from "local file" Browser Source to a "URL" browser source which points at the local file).  He also had the same issue where the "location" didn't change - he wanted the video to be in the middle, but it was always top left.

One thing I didn't mention before is that I'm using OBS 0.11.4-1872-ge6106015 which is an OBS with FTL Alpha build for Mixer.  My streamer friend was using the OBS Studio 18.0.1 & FTL 0.9.4.  Maybe that is part of the issue?

Unfortunately, I haven't been able to get the Troubleshooting to work to figure out why it's not rendering the page properly.

**Comment by ThePerry-2015** [comment]
That's the same version of OBS that I was using while I added the video function into Firebot.
When you have been switching around on the settings have you also reset the overlay between each change? OBS caches the Browser Source, and it doesn't bother to refresh that cache unless you force it to refresh, restart OBS completely or remove and re-add the Browser Source.

**Comment by imjeeves** [comment]
Went through @ThePerry-2015 tests:

Overlay compatibility: OBS

OBS overlay as local file: Local video, Position: Bottom Right, Size: 400x225 - **Does Not Play**
OBS overlay as local file: YouTube video, Position: Bottom Right, Size: 400x225 - **Does Not Play**
OBS as file:/// URL: Local video, Position: Bottom Right, Size: 400x225 - **Video Plays, but in Upper Left**
OBS as file/// URL: YouTube video, Position: Bottom Right, Size: 400x225 - **Video Plays, but in Upper Left**
Chrome: Local video, Position: Bottom Right, Size: 400x225 - **Does Not Play**
Chrome:: YouTube video, Position: Bottom Right, Size: 400x225 - **Video Plays, but in Upper Left**
Overlay Compatibility: Other

OBS as file:/// URL: Local video, Position: Bottom Right, Size: 400x225 - **Video Plays, but in Upper Left**
OBS as file/// URL: YouTube video, Position: Bottom Right, Size: 400x225 - **Video Plays, but in Upper Left**
OBS overlay as local file: Local video, Position: Bottom Right, Size: 400x225 - **Does Not Play**
OBS overlay as local file: YouTube video, Position: Bottom Right, Size: 400x225 - **Does Not Play**
Chrome: Local video, Position: Bottom Right, Size: 400x225 - **Video Plays, but in Upper Left**
Chrome:: YouTube video, Position: Bottom Right, Size: 400x225 - **Video Plays, but in Upper Left**

**NOTES**
So the two differences seem to be that, in OBS Compatibility mode, having the OBS Overlay as File "URL" with Local Video works for me just fine (though it's in the wrong position), and in "Other" compatibility mode, OBS Overlay as a local file with YouTube video did *not* work.  

Also, as noted, every time the video *did* play, it wasn't in the correct position.  I might try running the install wizard again - maybe just reinstalling Firebot completely to see if that fixes anything.

**Comment by ThePerry-2015** [comment]
@imjeeves 
I'm writing a custom overlay package for you that throws some log lines and stuff in combination with: https://github.com/Firebottle/Firebot/wiki/Troubleshooting-Firebot-Overlay-issues-in-OBS-Studio we should at least be able track down that reason for the positioning not working.
Give me a couple of minutes and I'll have an overlay-package ready for you. :)

**Comment by ThePerry-2015** [comment]
Go into  Settings -> Open Root Folder, open overlay -> js -> handlers. Rename the videoHandler.js file to something else, copy the videoHandler.js file that is inside the attached ZIP-file into the same folder and refresh your OBS Browser Source for firebot. And with the Developer Console Tool in the above Wiki page open in your browser try triggering your video-buttons.

This should output some log lines in the Developer Console Tool, either save this output as a file and zip it and attach to this thread, or copy and paste the console output from the developer console, put the pasted contents within ` ``` ` and ` ``` ` so it keeps the line returns as a code-block, makes it a lot easier to read.
It would also help if you can attach your controls file(user-settings -> controls -> name-of-your-board.json) so I can see the contents of that file, might have to zip it so you can attach it to the thread as well.

[videoHandler.zip](https://github.com/Firebottle/Firebot/files/1203480/videoHandler.zip)

**Comment by imjeeves** [comment]
@ThePerry-2015 
So when I have the overlay set to "Local File" literally *nothing* happens in the console.  

When I do the "URL", I go to the console, and the first error I see is "Failed to load resource: net::ERR_FILE_NOT_FOUND for "main.css".  I then am able to play one of my videos:

I couldn't find a way to copy/past the output from the console, so I took a screen cap of it and am attaching it here.

![image](https://user-images.githubusercontent.com/1455789/29007514-9081117e-7ab9-11e7-9d3d-633d90760abc.png)

Also attaching the JSON controls

[Firebot Board.zip](https://github.com/Firebottle/Firebot/files/1203496/Firebot.Board.zip)

One thing I noticed, for what it's worth, is the difference between the Elements loaded when I use the OBS Overlay as a "Local File" vs when I use it as an URL.  See below (not sure if this makes a difference, but it seemed odd that none of the important scripts/handlers were being loaded when I use the Local File.

Local File: 
![image](https://user-images.githubusercontent.com/1455789/29007535-118a6162-7aba-11e7-98d3-d47abc2298b8.png)

URL to the same file: 
![image](https://user-images.githubusercontent.com/1455789/29007541-3517e94c-7aba-11e7-9ebf-35a356cd4a4d.png)

**Comment by ThePerry-2015** [comment]
Hmm, that's odd.. Can you hover the css/main.css line in the `link href=` line there and take a screenshot of the URL it shows for that file? As well as check the contents of the overlay -> css folder.

**Comment by imjeeves** [comment]
so there's no css folder... only an "scss" folder.

There is a file in there called "main"... but it's "main.scss"

**Comment by ThePerry-2015** [comment]
Ok, we might have located an issue there with the latest Firebot package.. It seems someone might have forgot to run a script before packaging it and pushing the update. If you had 4.0.0 or 4.0.1 installed from earlier it would have been there, that's why it has passed by when we have tested and not seen the same error as you. Unzip this file into the overlay folder and you should get the main.css file.. :) Then the positioning should be fixed. I'm still doing some tests on the other stuff to see if that might be caused by the same mistake.

[css.zip](https://github.com/Firebottle/Firebot/files/1203511/css.zip)

**Comment by imjeeves** [comment]
Awesome, that fixes the positioning issue.

Oh, and it seems to have fixed the "local file" issue as well.  I'm able to play videos using the Local File OBS Overlay now.  SO that's cool.  Thanks for the help!

**Comment by ebiggz** [comment]
For Devs: We need to remember to run the scss compiler when making a release build to ensure the overlay css is generated and present.

**Comment by Firebottle** [comment]
Released 4.1.0. Let us know if anyone has any issues.
